opentelemetry/src/trace/context.rs

```diff
@@ -452,3 +454,151 @@ pub trait FutureExt: Sized {
         self.with_context(otel_cx)
     }
 }
+
+/// Lazily creates a new span only if the current context has an active span,
+/// which will used as the new span's parent.
+///
+/// This is useful for instrumenting library crates whose activities would be
+/// undesirable to see as root spans, by themselves, outside of any application
+/// context.
+///
+/// # Examples
+///
+/// ```
+/// use opentelemetry::trace::{new_span_if_active, RecruitedTracer, SpanBuilder};
+///
+/// fn my_lib_fn() {
+///     let _guard = new_span_if_active(
+///         || SpanBuilder::from_name("my span"),
+///         RecruitedTracer::lazy(&|| opentelemetry::global::tracer(module_path!())),
+///     )
+///     .map(|cx| cx.attach());
+/// }
+/// ```
+pub fn new_span_if_active(
```

Comment:
So all of `new_span_if_active()`, `new_span_if_active_else_current()` and `with_current_span()` build on top of `Context::map_current()` -- I think they're effectively "syntax sugar"? Do you think it's feasible/a good idea to build something more modular and/or make the core API more extensible so that it's easy to build this stuff on top?

Reply:
It might be feasible, but I'm not readily seeing the alternative options for making it more modular or extensible (without being verbose).